modules/basegl/src/processors/volumeprocessing/volumegradientprocessor.cpp

```diff
@@ -97,14 +97,17 @@ VolumeGradientProcessor::~VolumeGradientProcessor() = default;
 
 void VolumeGradientProcessor::preProcess(TextureUnitContainer&) {
     shader_.setUniform("channel", channel_.getSelectedValue());
+    shader_.setUniform("dataRange", vec2(inport_.getData()->dataMap_.dataRange));
 }
 
 void VolumeGradientProcessor::postProcess() {
-    volume_->dataMap_.valueAxis.name = "gradient";
-    volume_->dataMap_.valueAxis.unit =
-        inport_.getData()->dataMap_.valueAxis.unit / inport_.getData()->axes[0].unit;
-    volume_->dataMap_.dataRange = dvec2(-1.0, 1.0);
-    volume_->dataMap_.valueRange = dvec2(-1.0, 1.0);
+    if (!dataInChannel4_) {
```

Comment:
Since the fourth channel contains now float values in the original input range, we need to reuse the original data and value ranges here.

A slightly better solution would be to have a range mapping for each channel (and stop mixing up data format and actual data ranges).

Reply:
But that means that the gradients will be scaled incorrectly

Reply:
well otherwise the data will be scaled incorrectly...